# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/span.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/naga/src/span.rs
--- a/third_party/rust/naga/src/span.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/naga/src/span.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -6,7 +6,7 @@
 };
 use core::{error::Error, fmt, ops::Range};
 
-use crate::{error::replace_control_chars, path_like::PathLike, Arena, Handle, UniqueArena};
+use crate::{error::replace_control_chars, Arena, Handle, UniqueArena};
 
 /// A source code span, used for error reporting.
 #[derive(Clone, Copy, Debug, PartialEq, Default)]
@@ -283,14 +283,12 @@
 
     /// Emits a summary of the error to standard error stream.
     #[cfg(feature = "stderr")]
-    pub fn emit_to_stderr_with_path<P>(&self, source: &str, path: P)
+    pub fn emit_to_stderr_with_path(&self, source: &str, path: &str)
     where
         E: Error,
-        P: PathLike,
     {
         use codespan_reporting::{files, term};
 
-        let path = path.to_string_lossy();
         let files = files::SimpleFile::new(path, replace_control_chars(source));
         let config = term::Config::default();
 
@@ -315,14 +313,12 @@
     }
 
     /// Emits a summary of the error to a string.
-    pub fn emit_to_string_with_path<P>(&self, source: &str, path: P) -> String
+    pub fn emit_to_string_with_path(&self, source: &str, path: &str) -> String
     where
         E: Error,
-        P: PathLike,
     {
         use codespan_reporting::{files, term};
 
-        let path = path.to_string_lossy();
         let files = files::SimpleFile::new(path, replace_control_chars(source));
         let config = term::Config::default();
 